mergin/client.py

```diff
@@ -915,6 +925,58 @@ def pull_project(self, directory):
         pull_project_wait(job)
         return pull_project_finalize(job)
 
+    def sync_project(self, project_dir):
+        """
+        Syncs project by loop with these steps:
+        1. Pull server version
+        2. Get local changes
+        3. Push first change batch
+        Repeat if there are more changes pending.
+        """
+        has_more_changes = True
+        while has_more_changes:
+            self.pull_job = pull_project_async(self, project_dir)
+            pull_project_wait(self.pull_job)
+            pull_project_finalize(self.pull_job)
+
+            changes_handler = ChangesHandler(self, project_dir)
+            changes_batch, has_more_changes = changes_handler.get_change_batch()
+
+            self.push_job = push_project_async(project_dir, changes_batch)
+            push_project_wait(self.push_job)
+            push_project_finalize(self.push_job)
+
+    def sync_project_with_callback(self, project_dir, progress_callback=None, sleep_time=100):
+        """
+        Syncs project while sending push progress info as callback.
+        Sync is done in this loop:
+            Pending changes? -> Pull -> Push change batch -> repeat
```

Comment:
can you please add a docstring about how the `progress_callback` function should look like? (and what is the `sleep_time`)